bytecomp/bytelink.ml

```diff
@@ -503,9 +503,9 @@ let link_bytecode_as_c tolink outfile with_main =
        output_string outchan "\n};\n\n";
        (* The sections *)
        let sections =
-         [ Bytesections.Name.SYMB, Symtable.data_global_map();
-           Bytesections.Name.PRIM, Obj.repr(Symtable.data_primitive_names());
-           Bytesections.Name.CRCS, Obj.repr(extract_crc_interfaces()) ] in
+         [ Bytesections.Name.to_string SYMB, Symtable.data_global_map();
+           Bytesections.Name.to_string PRIM, Obj.repr(Symtable.data_primitive_names());
+           Bytesections.Name.to_string CRCS, Obj.repr(extract_crc_interfaces()) ] in
        output_string outchan "static char caml_sections[] = {\n";
        output_data_string outchan
          (Marshal.to_string sections []);
```

Comment:
Could you add a type annotation on `sections`? It could have forced us to notice the issue. (I think all magic boundaries should be type-annotated.)